File: 1207-unique-number-of-occurrences/1207--unique-number-of-occurrences.js
var uniqueOccurrences = function(arr) {
    // Create a map to count occurrences of each number
    const cnt = new Map();
    
    // Populate the map with counts
    for (const x of arr) {
        cnt.set(x, (cnt.get(x) || 0) + 1);
    }
    
    // Check if all occurrence counts are unique
    return cnt.size === new Set(cnt.values()).size;
};

File: 1207-unique-number-of-occurrences/1207--unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        // Count occurrences of each number
        Map<Integer, Integer> cnt = new HashMap<>();
        for (int x : arr) {
            cnt.merge(x, 1, Integer::sum);
        }
        
        // Check if all occurrence counts are unique
        return new HashSet<>(cnt.values()).size() == cnt.size();
    }
}
